### PR TITLE
Working with roles and environments

### PR DESCRIPTION
**Pull request by Durf1985**
# Сreate environments and roles

## In the process done:

* Created roles `app` and `db` with using `ansible-galaxy init`
* Transfer of old code in accordance with the structure of roles
* Created stage and prod environments
* Added dynamic inventory to each environment
* Installing a community role `jdauphant.nginx`
* Configuring `nginx` to work with `reddit monolith` as a reverse proxy
* Encryption of credentials with Ansible Vault
## How to launch a project:

Run command `ansible-playbook ~/your_repo/ansible/playbook/site.yml`

## How to check the performance:

In the browser, type in the address bar `http://app_vm_external_address:80`

## PR checklist

The label with the homework number, in label description - homework lecture and topic

### CODE REVIEW

File: ansible/roles/app/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+- name: Add unit file for Puma
+  ansible.builtin.copy:
+    src: files/monapp.service
+    dest: /etc/systemd/system/monapp.service
+  notify: Reload puma
+
+- name: Add config for DB connection
+  ansible.builtin.template:
+    src: templates/db_config.j2
```
Comment (loktionovam) [inline ansible/roles/app/tasks/main.yml]: Here you can use `db_config.j2` without `template` directory prefix (like you do when template `src: mongod.conf.j2` )

File: ansible/environments/prod/group_vars/app [hunk 2]
```diff
@@ -1 +1,7 @@
 db_host: "{{hostvars['reddit-db']['gce_private_ip']}}"
+nginx_sites:
+default:
```
Comment (loktionovam) [inline ansible/environments/prod/group_vars/app]: Here is a problem with indentation

**Review by loktionovam — CHANGES_REQUESTED** [review]

File: ansible/ansible.cfg [hunk 3]
```diff
@@ -9,6 +9,7 @@ remote_user = appuser
 private_key_file = ~/.ssh/appuser
 host_key_checking = False
 retry_files_enabled = False
+vault_password_file = ./vault.key
```
Comment (loktionovam) [inline ansible/ansible.cfg]: It is better to store vault key completely outside of your repo, for example in `~/.ansible/vault.key`. This prevents accidentally commit of your key.

**Review by loktionovam — CHANGES_REQUESTED** [review]

File: ansible/ansible.cfg (hunk 3 above)
Comment (Durf1985) [reply ansible/ansible.cfg]: fix

**Review by loktionovam — APPROVED** [review]

**Pull request merged**